src/librustc_typeck/diagnostics.rs

```diff
@@ -2617,6 +2774,22 @@ For more information see the [opt-in builtin traits RFC](https://github.com/rust
 -lang/rfcs/blob/master/text/0019-opt-in-builtin-traits.md).
 "##,
 
+E0390: r##"
+You tried to implement on an `*mut T` type. Erroneous code example:
+
+```
+struct Foo {
+    x: i32
+}
+
+impl *mut Foo {}
+// error: only a single inherent implementation marked with
+//        `#[lang = "mut_ptr"]` is allowed for the `*mut T` primitive
+```
+
+To fix this, please follow the compiler recommendations.
```

Comment:
The compiler doesn't recommend anything here.

How about "This isn't allowed, perhaps you might get the desired effects by wrapping the raw pointer in a struct", and then suggest doing `struct FooPtr(pub *mut Foo)` or something